src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -277,13 +277,19 @@ protected virtual bool IsAttributeConditionallyOmitted(NamedTypeSymbol attribute
                 ArrayBuilder<BoundExpression> boundNamedArgumentsBuilder = null;
                 HashSet<string> boundNamedArgumentsSet = null;
 
+                // VariableUsePass assumes we're not in a scope that can reference local variables.
+                // Need to rework VariableUsePass if that ever changes.
+                Debug.Assert((this.ContainingMemberOrLambda as MethodSymbol)?.MethodKind != MethodKind.LocalFunction);
```

Comment:
>Debug.Assert((this.ContainingMemberOrLambda as MethodSymbol)?.MethodKind != MethodKind.LocalFunction); [](start = 16, length = 102)

I am pretty sure this code is getting executed for attributes within local function declaration. This assert is not guarding against that. Obviously tests that were added are not adequate, they should be testing behavior of the compiler rather than trying to prove that certain asserts do not fire. Please start with tests that test expected behavior.